### PR TITLE
Add more description around imprecise bus faults

### PR DESCRIPTION
**Pull request by SenRamakri**
Add more description around imprecise bus faults and added more clarifications.

### CODE REVIEW

File: docs/tutorials/debug/debug_fault_exception.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ When the system crashes due to fault exceptions, the Mbed OS fault exception han
 
 The exceptions supported on your platform depend on the specific Cortex-M core you have in your system.
 
-For example, Cortex-M0 cores (or any ARMv6M cores) do not have MemManage, BusFault and UsageFault exceptions implemented. In those cases, all exceptions are reported as HardFault exception. Please look at the **Technical Reference Manual** and **Arm Architecture Reference Manual** documents for more information on exceptions supported for the specific core you have in your system.
+For example, Cortex-M0/M0+ processors (or any ARMv6M processors) do not have MemManage, BusFault and UsageFault exceptions implemented. In those cases, all exceptions are reported as HardFault exception. For ARMv7M processors, MemManage, BusFault and UsageFault exceptions are triggered only if they are enabled in **SHCSR** (System Handler Control and State Register). Please look at the **Technical Reference Manual** and **Arm Architecture Reference Manual** documents for more information on exceptions supported for the specific core you have in your system.
```
Comment (AnotherButler) [inline docs/tutorials/debug/debug_fault_exception.md]: Query: What triggers MemManage, BusFault and UsageFault exceptions? Please clarify for active voice.

File: docs/tutorials/debug/debug_fault_exception.md [hunk 2]
```diff
@@ -78,6 +78,10 @@ Note that the **LR** value may not reflect the actual caller, depending on the i
...
...M4 processors have write buffers, which are high-speed memory between the processor and main memory and whose purpose is to optimize stores to main memory. This is good for performance because the processor can proceed to the next instruction without having to wait for the write transaction to complete. However, this can cause imprecise bus faults in which the processor could have executed a number of instructions including branch instructions by the time bus fault is triggered. This makes it harder to debug imprecise faults because we cannot tell which instruction caused the fault. You can verify if you are encountering an imprecise fault by looking at the **BFSR.IMPRECISERR** (bit 2 of **BFSR**) status bit. To help debugging such situations, you can try disabling the write buffer by setting the **DISDEFWBUF** bit in **ACTLR** (Auxiliary Control Register), which makes those exceptions precise. Please look at the **Technical Reference Manual** and **Arm Architecture Reference Manual** documents for more information on fault exception types and information on these registers. Note that disabling the write buffer affects performance, so you may not want to do that in production code.
```
Comment (AnotherButler) [inline docs/tutorials/debug/debug_fault_exception.md]: Query: Who or what triggers the bus fault? You, the user?

**Comment by SenRamakri** [comment]
Hi @AnotherButler , Thanks for your review and I have addressed your comments by providing an explanation of Fault handlers. I have also made few other minor changes. Can you please review?

**Comment by SenRamakri** [comment]
Can you please add Kevin as well to the review?

**Pull request merged**